src/PrestaShopBundle/Translation/Provider/BackofficeCatalogueProvider.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShopBundle\Translation\Loader\DatabaseTranslationLoader;
+use Symfony\Component\Translation\MessageCatalogue;
+
+class BackofficeCatalogueProvider implements CatalogueProviderInterface
+{
+    /**
+     * We need a connection to DB to load user translated catalogue.
+     *
+     * @var DatabaseTranslationLoader
+     */
+    private $databaseTranslationLoader;
+
+    /**
+     * This is the directory where Default and FileTranslated translations are stored.
+     * For the Backoffice catalogue,
+     *   - Default catalogue is within resourceDirectory/default
+     *   - FileTranslated catalogue is in resourceDirectory/locale
+     *
+     * @var string
+     */
+    private $resourceDirectory;
+
+    /**
+     * @param DatabaseTranslationLoader $databaseTranslationLoader
+     * @param string $resourceDirectory
+     */
+    public function __construct(
+        DatabaseTranslationLoader $databaseTranslationLoader,
+        string $resourceDirectory
+    ) {
+        $this->databaseTranslationLoader = $databaseTranslationLoader;
+        $this->resourceDirectory = $resourceDirectory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultCatalogue(string $locale, bool $empty = true): MessageCatalogue
+    {
+        $provider = new DefaultCatalogueProvider(
```

Comment:
> keep the same parameters for BackofficeCatalogueProvider but build the sub providers right away in the constructor and store them in private fields

Warning: this means when class is created (for example by Symfony container upon use), we'll perform the operations needed to create this object... even if later we don't use it.

> if you don't want to create sub providers for nothin (in case their related method is never called) you can create private getters that check if private field is null and create the provider only then (then it's stored so that it's created only once)

Seems better then 😉  it creates the provider upon need, and dont rebuild it if not needed.

That being said, it might be important to check:
- do we expect to call `getDefaultCatalogue()` multiple times? 🤔 
- if we do expect it to be called, would it be useful to recreate the `DefaultCatalogueProvider`? for example if files on disk have changed, this can allow to acknowledge the latest changes in the files